test/Dialect/FIRRTL/blackbox-reader.mlir

```diff
@@ -1,5 +1,10 @@
-// RUN: circt-opt --pass-pipeline='firrtl.circuit(firrtl-blackbox-reader)' %s | FileCheck %s
+// RUN: split-file %s %t
+// RUN: sed 's?@DIR@?%t?' %t/Foo.mlir.orig > %t/Foo.mlir
+// RUN: circt-opt --pass-pipeline='firrtl.circuit(firrtl-blackbox-reader)' %t/Foo.mlir | FileCheck %t/Foo.mlir
```

Comment:
What's going on here is that I want to do text substitution inside the test because I need to have a `BlackBoxPathAnno` that has a path that is valid for this test.  If the path is invalid then this will error out.  I do this using `sed`, _but not doing it in place_ because there is no portable way to do `sed -i` that works on both macOS and Linux (macOS/BSD wants `sed -i ''` and Linux wants `sed -i` 🤷‍♀️ ).